pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/AbstractIndexType.java

```diff
@@ -48,8 +50,18 @@ public String getId() {
 
   @Override
   public Map<String, C> getConfig(TableConfig tableConfig, Schema schema) {
+    return getConfig(tableConfig, schema, null);
+  }
+
+  @Override
+  public Map<String, C> getConfig(TableConfig tableConfig, Schema schema, @Nullable String tier) {
+    if (tier != null) {
+      ColumnConfigDeserializer<C> deserializer =
+          _deserializersByTier.computeIfAbsent(tier, t -> createDeserializer(tier));
+      return deserializer.deserialize(tableConfig, schema);
+    }
     if (_deserializer == null) {
-      _deserializer = createDeserializer();
+      _deserializer = createDeserializer(null);
```

Comment:
So we are only caching the deserializer used when there is no tier? I think it would be better to never cache the deserializer or do it for all tiers.

For example, I would suggest to actually change ColumnCOnfigDeserializer in order to add the tier parameter there, so there would always be a single deserializer that is able to read any tier.